src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PreparedStatementTest.java

```diff
@@ -613,24 +611,18 @@ private void testStatementPoolingInternal(String mode) throws Exception {
 
                 // test updated value, should be 1 + 100 = 101
                 // although executeBatch() throws exception, update operation should be executed successfully.
-                try (ResultSet rs = con.createStatement()
+                try (Statement stmt = con.createStatement(); ResultSet rs = stmt
                         .executeQuery("select * from " + AbstractSQLGenerator.escapeIdentifier(tableName2) + "")) {
                     rs.next();
                     assertSame(101, rs.getInt(1));
                 }
             }
-        }
-
-        try (SQLServerConnection con = (SQLServerConnection) DriverManager.getConnection(connectionString)) {
-            if (mode.equalsIgnoreCase("bulkcopy")) {
```

Comment:
why was this logic removed :0

Reply:
I'm saving creating new connection. mode is determined above.